### PR TITLE
nearly finished caseinsensitivedict

### PR DESCRIPTION
**Pull request by gwriterk**
merge get method and test to master branch

### CODE REVIEW

File: test_caseinsensitivedict.py [hunk 1]
```diff
@@ -67,6 +67,16 @@ def test_copy(self):
         #     Return the value for key if key is in the dictionary, else
         #     default.  If default is not given, it defaults to None, so that
         #     this method never raises a KeyError.
+    def test_get(self):
+        myinstance = CaseInsensitiveDict()
+        myinstance['Kaisa'] = 5
+        self.assertIn('Kaisa', myinstance)
+        self.assertNotIn('Jhin', myinstance)
+        value = myinstance.get('Kaisa')
```
Comment (hopkiw) [inline test_caseinsensitivedict.py:74]: add a case where we `.get()` the string 'kaisa' (all lower case). it should work for a case-insensitive dictionary and in this case it should return 5.

File: test_caseinsensitivedict.py [hunk 2]
```diff
@@ -100,6 +110,13 @@ def test_keys(self):
         # popitem()
         #     Remove and return a (key, value) pair from the dictionary.  Pairs
         #     are returned in LIFO order.
+    def test_popitem(self):
```
Comment (hopkiw) [inline test_caseinsensitivedict.py:103]: FYI these methods are not working, so shouldn't be merged to the master branch, until they're done.

File: test_caseinsensitivedict.py [hunk 3]
```diff
@@ -122,7 +139,10 @@ def test_keys(self):
     def test_values(self):
```
Comment (hopkiw) [inline test_caseinsensitivedict.py:138]: this test method also not working yet

**Review by hopkiw — CHANGES_REQUESTED** [review]

File: test_caseinsensitivedict.py (hunk 2 above)
Comment (hopkiw) [reply test_caseinsensitivedict.py:111]: if these are working now, say so and resolve this comment

File: test_caseinsensitivedict.py [hunk 4]
```diff
@@ -67,10 +67,18 @@ def test_copy(self):
         #     Return the value for key if key is in the dictionary, else
         #     default.  If default is not given, it defaults to None, so that
         #     this method never raises a KeyError.
+    def test_get(self):
+        myinstance = CaseInsensitiveDict()
+        myinstance['Kaisa'] = 5
+        myinstance['kaisa'] = 5
```
Comment (hopkiw) [inline test_caseinsensitivedict.py]: what does this line do?

File: test_caseinsensitivedict.py [hunk 5]
```diff
@@ -67,10 +67,18 @@ def test_copy(self):
...
+        self.assertIn('Kaisa', myinstance)
+        self.assertNotIn('Jhin', myinstance)
+        value = myinstance.get('Kaisa')
+        nonvalue = myinstance.get('Jhin')
+        self.assertEqual(value, 5)
+        self.assertEqual(nonvalue, None)
+        self.assertEqual(myinstance['kaisa'], 5)
```
Comment (hopkiw) [inline test_caseinsensitivedict.py]: using the `name[key]` syntax will invoke a specific method in our class, but *not* use the `get()` method. do you remember what method it will use?

the purpose of `test_get()` is to test the `get()` method, so change this here

File: test_caseinsensitivedict.py [hunk 6]
```diff
@@ -100,12 +108,27 @@ def test_keys(self):
         # popitem()
         #     Remove and return a (key, value) pair from the dictionary.  Pairs
         #     are returned in LIFO order.
+    def test_popitem(self):
+        myinstance = CaseInsensitiveDict()
+        myinstance['Kaisa'] = 5
+        self.assertIn('Kaisa', myinstance)
+        item = myinstance.popitem()
+        self.assertEqual(item, ('Kaisa', 5))
+        self.assertNotIn(item, myinstance)
```
Comment (hopkiw) [inline test_caseinsensitivedict.py]: what kinds of things can we use the 'in' operator on a dictionary with? what kind of thing are you using here?

File: test_caseinsensitivedict.py [hunk 7]
```diff
@@ -100,12 +108,27 @@ def test_keys(self):
...
         #     If key is in the dictionary, return its value. If not, insert key
         #     with a value of default and return default.  default defaults to
         #     None.
 
+    def test_setdefault(self):
+        myinstance = CaseInsensitiveDict()
+        myinstance['Kaisa'] = 5
+        item = myinstance.setdefault('Kaisa')
+        self.assertEqual(item, 5)
+        default_item = myinstance.setdefault('Jhin')
```
Comment (hopkiw) [inline test_caseinsensitivedict.py]: `setdefault()` can take one or two parameters, but you're only testing the single parameter case

File: test_caseinsensitivedict.py (hunk 6 above)
Comment (hopkiw) [reply test_caseinsensitivedict.py]: can we try adding 3 items and then popping all 3 (testing each value as you go), and confirming it is empty at the end?

**Review by hopkiw — CHANGES_REQUESTED** [review]
